### PR TITLE
Feature/test improvements

### LINKED ISSUE #22: Implement mutation testing with mutmut
## Overview
Configure and implement mutation testing to identify weak tests that pass even when code is broken.

## Current State
- mutmut installed in dev dependencies
- No configuration in pyproject.toml
- Unknown mutation score baseline

## Tasks
- [ ] Add mutmut configuration to pyproject.toml
- [ ] Run initial mutation scan: `poetry run mutmut run`
- [ ] Analyze survivors: `poetry run mutmut results`
- [ ] Document baseline mutation score (expect 60-70% initially)
- [ ] Fix weak tests to improve mutation score to 80%+
- [ ] Add GitHub workflow for mutation testing on PRs
- [ ] Add mutation score badge to README

## Configuration
```toml
[tool.mutmut]
paths_to_mutate = "souschef/"
backup = false
runner = "poetry run pytest -x --assert=plain -q"
tests_dir = "tests/"
```

## Success Criteria
- [ ] Mutation testing configured and running
- [ ] Baseline mutation score documented
- [ ] At least 80% mutation score achieved
- [ ] CI/CD integration complete

## References
- See TESTING_ROADMAP.md section 1
- Mutmut docs: https://mutmut.readthedocs.io/

## Effort
2-3 days

### PR DESCRIPTION
**Pull request by kpeacocke**
# Pull Request Template

## 🌿 Gitflow Branch Check

**Source branch**: `[your branch name]`
**Target branch**: `[base branch]`

Please verify your branch follows gitflow conventions:
- ✅ `feature/*` → targets `develop`
- ✅ `bugfix/*` → targets `develop`
- ✅ `release/*` → targets `main`
- ✅ `hotfix/*` → targets `main`

_The gitflow workflow will automatically validate this._

---

## 📋 Summary

Brief description of what this PR does and why.

**Type of Change** (check all that apply):
- [ ] 🐛 Bug fix (non-breaking change that fixes an issue)
- [ ] ✨ New feature (non-breaking change that adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📚 Documentation update
- [x] 🧪 Test improvements
- [x] 🔧 Refactoring (no functional changes)
- [x] 📦 Dependencies update
- [ ] 🚀 Performance improvement

## 🎯 Related Issues

Fixes #22 #23 #24 #25 #26 

### Core Changes
- No new tools, just lots of test updates

### Test Coverage
- [x] Unit tests added/updated for new functionality
- [x] Integration tests added/updated
- [x] Property-based tests added if applicable
- [x] All existing tests pass
- [x] Code coverage maintained or improved

### Manual Testing
Describe how you tested this change:

```bash
# Commands run for testing
souschef-cli recipe examples/recipe.rb
```

**Test Results**:
```
# Paste relevant test output
Resource 1: package[nginx]
  Type: package
  Action: install
```

### Test Cases Covered
- [x] Normal/happy path scenarios
- [x] Edge cases and error conditions
- [x] Cross-platform compatibility
- [x] Performance with large inputs
- [x] Integration with existing features

## 📊 Performance Impact

If applicable, describe performance implications:

- **Memory usage**: No significant change / Reduced / Increased by X%
- **Processing time**: Benchmarks show X% improvement/regression
- **Scalability**: Handles larger cookbooks / No impact

### Checklist (verify before submitting)
- [x] ✅ Code passes linting
- [x] ✅ Code is properly formatted
- [x] ✅ All tests pass
- [x] ✅ Type hints are complete and accurate
- [x] ✅ Docstrings follow Google style format
- [x] ✅ No warnings introduced (zero warnings policy)
- [x] ✅ Code follows existing patterns and style
- [x] ✅ Error handling is comprehensive

### Code Review Areas
Please pay special attention to:
- [ ] Resource-to-task conversion logic
- [ ] ERB-to-Jinja2 template conversion
- [ ] File path handling and security
- [ ] Error messages and user experience
- [x] Test coverage and edge cases

## 🔄 Reviewer Guidance

**Focus Areas for Review**:
- Code correctness and edge case handling
- Test coverage and quality
- Documentation accuracy
- Performance implications
- Security considerations

**Questions for Reviewers**:
- Is the Chef-to-Ansible mapping accurate?
- Are there edge cases I missed?
- Is the error handling appropriate?
- Should this be split into smaller PRs?

---

## 📋 Pre-Submit Checklist

Before submitting this PR, I have verified:

- [x] 🧪 All tests pass locally (`uv run pytest`)
- [x] 🎨 Code is linted and formatted (`uv run ruff check . && uv run ruff format .`)
- [x] 📊 Test coverage is maintained or improved
- [x] 📖 Documentation is updated where needed
- [x] 🔍 I have self-reviewed my code changes
- [x] 💭 Complex logic has explanatory comments
- [x] 🚨 No debug code or console.log statements left
- [x] 🔧 Commit messages follow conventional format

**By submitting this PR, I confirm that**:
- [x] My contribution is made under the project's MIT license
- [x] I have followed the project's coding standards and guidelines
- [x] I have read and agree to the [Code of Conduct](../CODE_OF_CONDUCT.md)

### CODE REVIEW

File: tests/test_integration_accuracy.py [hunk 1]
```diff
@@ -0,0 +1,591 @@
...
+        }"""
+        result = convert_chef_databag_to_vars(
+            databag_content=databag_content, databag_name="app", item_name="config"
+        )
+
+        # Verify nested structure preserved
+        assert "app" in result or "name" in result
+        assert "myapp" in result
+        assert "database" in result or "config" in result
+        assert "db.example.com" in result
```
[the remark this answers is not in the document]
Comment (kpeacocke) [reply tests/test_integration_accuracy.py]: Investigating

[the remark this answers is not in the document]
Comment (kpeacocke) [reply tests/test_integration_accuracy.py]: This is a data validation check in test code, not an attempt at URL security/sanitization. The warning was a false positive because the code wasn't trying to sanitize URLs - it was simply verifying that a data conversion function correctly preserved hostnames in its output.
Have added comment to code
